build.gradle.kts

```diff
@@ -9,6 +14,75 @@ allprojects {
     group = findProperty("GROUP").toString()
 }
 
+subprojects {
+    tasks.withType<JavaCompile>().configureEach {
+        sourceCompatibility = JavaVersion.VERSION_17.toString()
+        targetCompatibility = JavaVersion.VERSION_17.toString()
+    }
+
+    tasks.withType<KotlinCompile>().configureEach {
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_17.toString()
+        }
+    }
+
+    plugins.withType<BasePlugin>().configureEach {
```

Comment:
Android's `ApplicationPlugin` and `LibraryPlugin` have the base of `BasePlugin`. Likewise, `ApplicationExtension` and `LibraryExtension` have the base of `BaseExtension`. So this code is saying that whenever we encounter a subproject where these plugins and extensions are applied, then we apply the following properties and logic there.

This has the effect of consistently applying the same config to all subprojects of these types.